src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -158,6 +160,35 @@ internal static bool VerifyRestoreOutput(RestoreRequest request, CacheFile cache
                 return false;
             }
 
+            if (request.UpdatePackageLastAccessTime)
+            {
+                foreach (var package in cacheFile.ExpectedPackageFilePaths)
+                {
+                    var packageFolder = new DirectoryInfo(Directory.GetParent(package).FullName).Parent.Parent.FullName;
+
+                    // we avoid fallback folders as they are readonly
+                    if (request.Project.RestoreMetadata.FallbackFolders.Any(fallbackFolder => PathUtility.GetStringComparerBasedOnOS().Equals(fallbackFolder, packageFolder)))
+                        continue;
+
+                    var packageParent = Directory.GetParent(package);
+                    var packageId = new DirectoryInfo(packageParent.FullName).Parent.Name;
+                    var version = packageParent.Name;
+                    var resolver = new VersionFolderPathResolver(packageFolder);
+                    var metadataFile = resolver.GetNupkgMetadataPath(packageId, new NuGetVersion(version));
```

Comment:
I think this should do fewer allocations, and therefore have higher performance.

```suggestion
                    var packageDirectory = Path.GetDirectory(package);
                    var metadataFile = Path.Combine(package, PackagingCoreConstants.NupkgMetadataFileExtension);
```